components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1204,6 +1216,49 @@ def wrap(request, *args, **kwargs):
     return wrap
 
 
+def json_response(f):
```

Comment:
In order to not repeat the mistakes of `@jsonp` and have no decorator configurability going forward let's switch this to a class based decorator like `login_required()`. To do that have an empty constructor, put your `wrap()` into a `__call__()` method, switch your decorator usage to `@json_response()`, and you should be set. I'm not 100% on the interplay with `@wraps` and the current consensus on the best way to do this but you should be able use `update_wrapper` from `functools` as in `@login_required()` as an example.